client/components/UserMenu.js

```diff
@@ -15,84 +15,137 @@ import {
   withCurrentUserId,
 } from '../store/selectors'
 import { Link } from '../routes'
-import AdminUserMenuNav from './AdminUserMenuNav'
 import userHasPermission from '../helpers/userHasPermission'
 
 const UserMenu = (props) => {
+  const {
+    showRescueList,
+    showUserList,
+  } = props
+
   const {
     authenticatedPage,
     loggedIn,
     logout,
     user,
     userAvatar,
     userId,
-    showAdmin,
   } = props
 
+  const renderNavItem = (item) => {
+    const {
+      action,
+      className,
+      key,
+      route,
+      routeParams,
+      permission,
+      title,
+    } = item
+
+    if (typeof (permission) === 'undefined' || permission) {
+      return (
+        <li className={className} key={key}>
+          <Link route={route} params={routeParams}>
+            <a {...(action && { href: '#', onClick: action })}>
+              <span>{title}</span>
+            </a>
+          </Link>
+        </li>
+      )
+    }
+
+    return null
+  }
+
+  const renderNav = (nav) => {
+    const {
+      header,
+      items,
+    } = nav
+
+    const permissions = items.map((item) => item.permission)
+    const permitted = permissions.includes(true) || permissions.includes(undefined)
```

Comment:
This is kinda inefficient since we have to iterate over every entry 3 times.

perhaps something like:

```javascript
    const permitted = items.filter((item) => typeof item.permission === 'undefined' || item.permission).length > 0
```

would work better